torch/csrc/jit/codegen/cuda/compute_at.cpp

```diff
@@ -506,6 +506,82 @@ void ComputeAt::traverseForward() {
   }
 }
 
+namespace {
+
+unsigned int getInnermostNonBroadcastIdFrom(TensorView* tv, unsigned int from) {
+  TORCH_INTERNAL_ASSERT(from <= tv->nDims());
+
+  int ret = from;
+
+  // Still assuming we only have block broadcast for now.
+  //  This part may change
+  while (ret > 0 && tv->axis(ret - 1)->isBroadcast()) {
+    ret--;
+  }
+
+  return ret;
+}
+
+} // namespace
+
+void ComputeAt::hoistInnermostBroadcast() {
+  // Get the dependency chains of values that could have
+  //  been updated by computeAt pass
+  auto chains =
+      tvChains(DependencyCheck::getAllDependencyChains(producer_, consumer_));
+
+  std::unordered_set<TensorView*> updated_producer;
+  std::unordered_set<TensorView*> visited_producer;
+
+  for (auto tv_chain : chains) {
```

Comment:
Can you avoid going through all the chains? I would recommend creating a set of all TVs. Check them all for the computeAt vs broadcast condition. Reduce them one by one, add all their consumers to another set, then do a pass on all the consumers to reset the produceAt position.

Chains can be very redundant and seems like it could also simplify the code.

Reply:
Makes sense. Updated to not traverse the chains. Thanks!